cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaControllerConfigurationDiff.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.operator.cluster.operator.resource;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.fabric8.zjsonpatch.JsonDiff;
+import io.strimzi.operator.common.model.OrderedProperties;
+import io.strimzi.operator.common.operator.resource.AbstractJsonDiff;
+
+import java.util.List;
+import java.util.Map;
+
+public class KafkaControllerConfigurationDiff extends AbstractJsonDiff {
+    final boolean configsHaveChanged;
+
+    protected KafkaControllerConfigurationDiff(Map<String, String> controllerConfig, String desired) {
```

Comment:
when I was thinking about how to pass the `controllerConfig` I was wonder how much was difficult to build a `Config` instance as we have in the current `KafkaBrokerConfigurationDiff` class. Of course, it's working there because `Config` (coming from Apache Kafka) is returned by the description via the admin client. The idea I had was to build a `Config` instance from what I have got via the Kafka Agent as a JSON. This would have allowed to not change this diffing class when the admin client API for controllers work (so returning a `Config` instance). Of course, building a `Config` instance means adding `ConfigEntry` which could be difficult to create based on how many info we need to know (readonly, default, ...) but I was thinking about referring to the model for that.
It was just an idea I had to have this diffing code to be reusable as much as possible but didn't go through enough to say ... it's crazy and not feasible. Maybe something you could think about.